pennylane/templates/subroutines/qft.py

```diff
@@ -61,7 +61,70 @@ def circuit_qft(basis_state):
             qml.QFT(wires=range(wires))
             return qml.state()
 
-        circuit_qft(np.array([1.0, 0.0, 0.0], requires_grad=False))
+    .. code-block:: pycon
+
+        >>> circuit_qft(np.array([1.0, 0.0, 0.0]))
+        [ 0.35355339+0.j -0.35355339+0.j  0.35355339+0.j -0.35355339+0.j
+          0.35355339+0.j -0.35355339+0.j  0.35355339+0.j -0.35355339+0.j]
+
+    .. details::
+        :title: Semiclassical Quantum Fourier transform
```

Comment:
Should this be a standalone demo? 🤔

Reply:
We could have it in here and it can be used as a seed for a demo later?

Reply:
Nice! @trbromley probably backlog given the other content we have in the works but might be good to note somewhere

Reply:
Agreed, this could be a short and sweet how-to guide. We could even think about adding a template for the semiclassical QFT.

For now, happy to have this here. But if we do add demo content, we can remove the content here and instead point to the demo.